### PR TITLE
Use IHttpClientFactory.CreateClient() instead of static HttpClient

### LINKED ISSUE #15346: use IHttpClientFactory in MediaStep to create instance of HttpClient
The `MediaStep` creates a static HttpClient instance. I think we should be using IHttpFactory there instead,

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #15346

### CODE REVIEW

**Comment by hishamco** [comment]
I will check the unit tests seems something broken due the injection

File: src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ExportRemoteInstanceController.cs [hunk 1]
```diff
@@ -35,13 +34,15 @@ public class ExportRemoteInstanceController : Controller
...
             INotifier notifier,
+            IHttpClientFactory httpClientFactory,
             IHtmlLocalizer<ExportRemoteInstanceController> localizer)
         {
             _authorizationService = authorizationService;
             _deploymentManager = deploymentManager;
             _session = session;
             _service = service;
             _notifier = notifier;
+            _httpClient = httpClientFactory.CreateClient();
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ExportRemoteInstanceController.cs:45]: Don't create it in the constructor.  Remove the global variable and create one before calling it in the action method.

File: src/OrchardCore.Modules/OrchardCore.Media/Recipes/MediaStep.cs [hunk 2]
```diff
@@ -21,15 +21,17 @@ public class MediaStep : IRecipeStepHandler
...
 
         public MediaStep(
             IMediaFileStore mediaFileStore,
             IOptions<MediaOptions> options,
+            IHttpClientFactory httpClientFactory,
             ILogger<MediaStep> logger)
         {
             _mediaFileStore = mediaFileStore;
             _allowedFileExtensions = options.Value.AllowedFileExtensions;
+            _httpClient = httpClientFactory.CreateClient();
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Media/Recipes/MediaStep.cs:34]: Same here. Done create it in the constructor. Instead just before you use it.

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ExportRemoteInstanceController.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ExportRemoteInstanceController.cs:45]: I thought about that, but it's fine to move it when it's needed to create the instance

**Pull request merged**